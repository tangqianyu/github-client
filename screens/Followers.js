import * as React from "react";
export default class Following extends React.Component {
  render() {
    return null;
  }
}
