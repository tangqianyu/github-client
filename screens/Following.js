import * as React from "react";
export default class Followers extends React.Component {
  render() {
    return null;
  }
}
