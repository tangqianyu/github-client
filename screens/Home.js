import React from "react";
import { StyleSheet, View, Text, Image, TouchableOpacity } from "react-native";
import { useQuery, gql } from '@apollo/client';
import Profile from "../models/profile";

export const PROFILE_QUERY = gql`
  query {
    viewer {
      name
      avatarUrl
      login
      websiteUrl
      bio
      email
      repositories {
        totalCount
      }
      followers {
        totalCount
      }
      following {
        totalCount
      }
    }
  }
`;

export const Home = ({ navigation }) => {
  const { loading, error, data } = useQuery(PROFILE_QUERY);
  if (loading) {
    return (
      <View style={styles.container}>
        <Text>Loading...</Text>
      </View>
    );
  }
  const profile = new Profile(data.viewer);
  return (
    <View style={styles.container}>
      <Text style={styles.header}>{profile.username}'s GitHub !</Text>
      <Image
        style={styles.avatar}
        source={{
          uri: profile.avatarUrl,
        }}
      />
      <View>
        <Text>Name</Text>
        <Text>{profile.nickname}</Text>
      </View>
      <View>
        <Text>Bio</Text>
        <Text>{profile.bio}</Text>
      </View>
      <View>
        <Text>Email</Text>
        <Text>{profile.email}</Text>
      </View>
      <View>
        <Text>website</Text>
        <Text>{profile.websiteUrl}</Text>
      </View>
      <View style={styles.bottom}>
        <TouchableOpacity
          style={styles.section}
          onPress={() => {
            navigation.navigate("Repos");
          }}
        >
          <Text>Repositories</Text>
          <Text>{profile.repositoriesCount}</Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={() => {
            navigation.navigate("Followers");
          }}
        >
          <Text>Followers</Text>
          <Text>{profile.followersCount}</Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={() => {
            navigation.navigate("Following");
          }}
        >
          <Text>Following</Text>
          <Text>{profile.followingCount}</Text>
        </TouchableOpacity>
      </View>
    </View>
  )
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  avatar: {
    width: 100,
    height: 100,
  },
  header: {
    fontWeight: "bold",
    fontSize: 26,
  },
  repoContainer: {
    height: 200,
    width: "100%",
  },
  bottom: {
    height: 100,
    flexDirection: "row",
    width: "100%",
  },
  section: {
    flex: 1,
  },
});


