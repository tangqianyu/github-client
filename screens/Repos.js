import React from "react";
import { StyleSheet, View, Text } from "react-native";
import { useQuery, gql } from '@apollo/client';
import RepoComponent from "../components/repo";
import { ScrollView } from "react-native";

const ALL_REPOS_QUERY = gql`
  query {
    viewer {
      repositories(last: 100) {
        nodes {
          name
          description
          owner {
            login
          }
        }
      }
    }
  }
`;


const Repo = () => {
  const { loading, error, data } = useQuery(ALL_REPOS_QUERY);
  if (loading) return (
    <View>
      <Text>Loading...</Text>
    </View>)

  return (
    <ScrollView style={styles.container}>
      {!loading &&
        data.viewer.repositories.nodes.map((repo, index) => (
          <RepoComponent repo={repo} key={index} />
        ))}
    </ScrollView>
  )
}



const styles = StyleSheet.create({
  container: {
    // flex: 1,
    // alignItems: "center",
    // justifyContent: "center",
  },
  header: {
    fontWeight: "bold",
    fontSize: 26,
  },
  repoContainer: {
    height: 200,
    width: "100%",
  },
});

export default Repo;
