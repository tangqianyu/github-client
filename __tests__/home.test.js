import { MockedProvider } from '@apollo/client/testing';
import React from 'react'
import { Home } from '../screens/Home'
import renderer from 'react-test-renderer';
import { PROFILE_QUERY } from '../screens/Home'


// it('should render loading state initially', () => {
//   const component = renderer.create(
//     <MockedProvider mocks={[]}>
//       <Home />
//     </MockedProvider>
//   );

//   const tree = component.toJSON();
//   expect(tree).toMatchInlineSnapshot(`
// <View
//   style={
//     Object {
//       "alignItems": "center",
//       "flex": 1,
//       "justifyContent": "center",
//         }
//   }
//   >
//   <Text>
//     Loading...
//   </Text>
// </View>`);
// });


it('should render Home', async () => {
  const profileMock = {
    request: {
      query: PROFILE_QUERY,
    },
    result: {
      data: {
        viewer: {
          __typename: "User",
          avatarUrl: "testAvatar",
          bio: "work hard!",
          email: "mengqiang.q@gmail.com",
          followers: {
            __typename: "FollowerConnection",
            totalCount: 17,
          },
          following: {
            __typename: "FollowingConnection",
            totalCount: 73,
          },
          login: "menq",
          name: "For177",
          repositories: {
            __typename: "RepositoryConnection",
            totalCount: 89,
          },
          websiteUrl: "testUrl",
        },
      }
    },
  };
  const wrapper = renderer.create(
    <MockedProvider mocks={[profileMock]}>
      <Home />
    </MockedProvider>
  );

  await new Promise(resolve => setTimeout(resolve, 0))

  /* toMatchSnapshot don't always work,it seems a bug */
  expect(wrapper.children).toMatchSnapshot()
});

