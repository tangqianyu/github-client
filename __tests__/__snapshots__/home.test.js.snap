// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 26,
        "fontWeight": "bold",
      }
    }
  >
    menq
    's GitHub !
  </Text>
  <Image
    source={
      Object {
        "uri": "testAvatar",
      }
    }
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  />
  <View>
    <Text>
      Name
    </Text>
    <Text>
      For177
    </Text>
  </View>
  <View>
    <Text>
      Bio
    </Text>
    <Text>
      work hard!
    </Text>
  </View>
  <View>
    <Text>
      Email
    </Text>
    <Text>
      mengqiang.q@gmail.com
    </Text>
  </View>
  <View>
    <Text>
      website
    </Text>
    <Text>
      testUrl
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 100,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <Text>
        Repositories
      </Text>
      <Text>
        89
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text>
        Followers
      </Text>
      <Text>
        17
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text>
        Following
      </Text>
      <Text>
        73
      </Text>
    </View>
  </View>
</View>
`;
